src/solvers/base_solver.hpp

```diff
@@ -30,18 +30,18 @@ class BaseSolver {
   /**
    * @brief Empty constructor
    *
-   * @param[in] comm MPI communicator
+   * @param[in] pmesh The primary mesh
    */
-  BaseSolver(MPI_Comm comm);
+  BaseSolver(std::shared_ptr<mfem::ParMesh> pmesh);
 
   /**
    * @brief Constructor that creates n entries in state_ of order p
    *
-   * @param[in] comm MPI communicator
+   * @param[in] pmesh The primary mesh
```

Comment:
p for primary or parallel is a tiny bit confusing.

Reply:
This is an unfortunately MFEM idiom I picked up. Since we don't use the serial flavor of meshes, we should just call it `mesh`.